include/RAJA/policy/desul/atomic_desul_openmp.hpp

```diff
@@ -0,0 +1,200 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2016-21, Lawrence Livermore National Security, LLC
+// and RAJA project contributors. See the RAJA/COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#ifndef RAJA_policy_atomic_desul_HPP
+#define RAJA_policy_atomic_desul_HPP
+
+#include "RAJA/config.hpp"
+
+#if defined(RAJA_ENABLE_OPENMP)
+
+#include "RAJA/util/macros.hpp"
+
+#include "RAJA/policy/atomic_builtin.hpp"
+
+#include "desul/atomics.hpp"
+
+// Default desul options for RAJA
+using raja_default_desul_order = desul::MemoryOrderRelaxed;
+using raja_default_desul_scope = desul::MemoryScopeDevice;
+
+namespace RAJA
+{
+
+#if defined(RAJA_COMPILER_MSVC)
+
+// For MS Visual C, just default to builtin_atomic for everything
+using omp_atomic = builtin_atomic;
+
+#else  // not defined RAJA_COMPILER_MSVC
+
+struct omp_atomic {};
+
+RAJA_SUPPRESS_HD_WARN
+template <typename T>
+RAJA_HOST_DEVICE
+RAJA_INLINE T
+atomicAdd(omp_atomic, T volatile *acc, T value) {
```

Comment:
So the atomic_desul_openmp file acts as the middle-man between the RAJA and desul APIs. At the moment RAJA uses this ```T volatile *acc``` parameter, but the desul API doesn't like ```volatile```.

Reply:
Right, volatile is orthogonal to the way that atomic works these days, and we decided not to use it.  Ideally it would be best if we could remove `volatile` from the raja interface, but if we have to leave it there, the `acc` variable will have to go through either a `const_cast<std::remove_volatile_t<decltype(*acc)>*>(acc)` or similar to cast off the `volatile`.

Reply:
Perhaps a 'dumb' question, but is there an advantage or a requirement to use ```std::remove_volatile_t``` as opposed to a simple ```const_cast<T*>(acc)``` to remove ```volatile``` in this case? So far I've only made the latter work for compilation, but I suspect that there may be something else going on behind the scenes that results in failed tests.

Reply:
Not really.  The `const_cast` should be fine here.  I do think that trying a switch to not taking volatile in the first place would be a good idea though.  If we can get away with it, the result will be more likely to behave correctly if someone passes in a pointer to a variable that was declared volatile than if we cast it away from the pointer unconditionally.

Reply:
Makes sense, the undefined behavior of casting volatile away might be causing some issues in testing from what I can gather. Moving the current atomic build away from volatile seems like the way to go long-term, so I'll tinker around and see if I can get that to work.